problems/first-non-repeated-character/first-non-repeating-character.kt

```diff
@@ -0,0 +1,15 @@
+fun firstNonRepeatedCharacter(input: String): Char? {
+    val seen = LinkedHashSet<Char>()
+    val seenOnce = LinkedHashSet<Char>()
+
+    input.forEach {                 // O(n)
+        if (seen.contains(it)) {    // O(1)
+            seenOnce.remove(it)     // O(1)
```

Comment:
We could also simply `return it` here, right? This way we could avoid declaring `seenOnce` if I'm not mistaken.

Reply:
Your solution is of course more generic, because it detects all duplicates. For the special case of finding the first non-repeating character, it's not really necessary, though. Just interested in your reasoning behind this. 😉

Reply:
You cannot return at this point because you might still encounter a repetition of the character that comes next in `seenOnce`.

1. let input be `abcab`
2. we add `a` to `seen` and `seenOnce`
3. we add `b` to `seen` and `seenOnce`
4. we add `c` to `seen` and `seenOnce`
5. we pop `a` from `seenOnce`

At this stage you’d want to return `b` and it would be wrong because `c` is actually the first non-repeated character but since we never looked at the whole `input` we couldn’t know.

Reply:
Ah right. *Non*-repeated character. 😅